src/components/swap/SwapModalFooter.tsx

```diff
@@ -1,36 +1,130 @@
+  decimalPlace: number
+): number => parseFloat(intialNumberObject.toFixed(decimalPlace))
+
+interface AnalyticsEventProps {
+  trade: InterfaceTrade<Currency, Currency, TradeType>
+  txHash: string | undefined
+  allowedSlippage: Percent
+  transactionDeadlineSecondsSinceEpoch: number | undefined
+  isAutoSlippage: boolean
+  isAutoRouterApi: boolean
+  tokenInAmountUsd: string | undefined
+  tokenOutAmountUsd: string | undefined
+  lpFeePercent: Percent
+}
+
+const formatAnalyticsEventProperties = ({
+  trade,
+  txHash,
+  allowedSlippage,
+  transactionDeadlineSecondsSinceEpoch,
+  isAutoSlippage,
+  isAutoRouterApi,
+  tokenInAmountUsd,
+  tokenOutAmountUsd,
+  lpFeePercent,
+}: AnalyticsEventProps) => ({
+  estimated_network_fee_usd: trade.gasUseEstimateUSD
+    ? getNumberFormattedToDecimalPlace(trade.gasUseEstimateUSD, 2)
+    : undefined,
+  transaction_hash: txHash,
+  transaction_deadline_seconds: getDurationTillTimestampSinceEpoch(transactionDeadlineSecondsSinceEpoch),
+  token_in_amount_usd: tokenInAmountUsd ? parseFloat(tokenInAmountUsd) : undefined,
+  token_out_amount_usd: tokenOutAmountUsd ? parseFloat(tokenOutAmountUsd) : undefined,
+  token_in_address: trade.inputAmount.currency.isToken ? trade.inputAmount.currency.address : undefined,
+  token_out_address: trade.outputAmount.currency.isToken ? trade.outputAmount.currency.address : undefined,
+  token_in_symbol: trade.inputAmount.currency.symbol,
+  token_out_symbol: trade.outputAmount.currency.symbol,
+  token_in_amount: getNumberFormattedToDecimalPlace(trade.inputAmount, trade.inputAmount.currency.decimals),
+  token_out_amount: getNumberFormattedToDecimalPlace(trade.outputAmount, trade.outputAmount.currency.decimals),
+  price_impact_percentage: getNumberFormattedToDecimalPlace(getPriceImpact(lpFeePercent, trade), 2),
+  allowed_slippage_percentage: getNumberFormattedToDecimalPlace(allowedSlippage, 2),
+  is_auto_router_api: isAutoRouterApi,
+  is_auto_slippage: isAutoSlippage,
+  chain_id:
+    trade.inputAmount.currency.chainId === trade.outputAmount.currency.chainId
+      ? trade.inputAmount.currency.chainId
+      : undefined,
+  // TODO(lynnshaoyu): implement duration_from_first_quote_to_swap_submission_seconds
+})
+
 export default function SwapModalFooter({
+  trade,
+  allowedSlippage,
+  txHash,
   onConfirm,
   swapErrorMessage,
   disabledConfirm,
 }: {
-  trade: Trade<Currency, Currency, TradeType>
+  trade: InterfaceTrade<Currency, Currency, TradeType>
+  txHash: string | undefined
+  allowedSlippage: Percent
   onConfirm: () => void
   swapErrorMessage: ReactNode | undefined
   disabledConfirm: boolean
 }) {
+  const transactionDeadlineSecondsSinceEpoch = useTransactionDeadline()?.toNumber() // in seconds since epoch
+  const isAutoSlippage = useUserSlippageTolerance() === 'auto'
+  const [clientSideRouter] = useClientSideRouter()
+  const tokenInAmountUsd = useStablecoinValue(trade.inputAmount)?.toFixed(2)
```

Comment:
same thing, i think we'll want to use the same logic we use around the rest of the App (sometimes having decimals beyond 2 matters even for USD)

Reply:
same as above.